pkg/cmd/checkendpoint/controller/controller.go

```diff
@@ -0,0 +1,237 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	operatorcontrolplanev1alpha1 "github.com/openshift/api/operatorcontrolplane/v1alpha1"
+	operatorcontrolplaneclientv1alpha1 "github.com/openshift/client-go/operatorcontrolplane/clientset/versioned/typed/operatorcontrolplane/v1alpha1"
+	alpha1 "github.com/openshift/client-go/operatorcontrolplane/informers/externalversions/operatorcontrolplane/v1alpha1"
+	"github.com/openshift/client-go/operatorcontrolplane/listers/operatorcontrolplane/v1alpha1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/checkendpoint/operatorcontrolplane/podnetworkconnectivitycheck/v1alpha1helpers"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/checkendpoint/trace"
+)
+
+type PodNetworkConnectivityCheckController interface {
+	factory.Controller
+}
+
+func New(podName, podNamespace string, checksGetter operatorcontrolplaneclientv1alpha1.PodNetworkConnectivityChecksGetter, checkInformer alpha1.PodNetworkConnectivityCheckInformer, recorder events.Recorder) PodNetworkConnectivityCheckController {
+	c := &controller{
+		podName:      podName,
+		podNamespace: podNamespace,
+		checksGetter: checksGetter.PodNetworkConnectivityChecks(podNamespace),
+		checkLister:  checkInformer.Lister().PodNetworkConnectivityChecks(podNamespace),
+		recorder:     recorder,
+		updaters:     map[string]StatusUpdater{},
+	}
+	c.Controller = factory.New().
+		WithSync(c.Sync).
+		WithInformers(checkInformer.Informer()).
+		ResyncEvery(10*time.Second).
+		ToController("check-endpoint", recorder)
+	return c
+}
+
+func (c *controller) Sync(ctx context.Context, syncContext factory.SyncContext) error {
+	checkList, err := c.checkLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	// filter list of checks for current pod
+	var checks []*operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck
+	for _, check := range checkList {
+		if check.Spec.SourcePod == c.podName {
+			checks = append(checks, check)
+		}
+	}
+
+	// create & start status updaters if needed
+	for _, check := range checks {
+		if updater := c.updaters[check.Name]; updater == nil {
+			c.updaters[check.Name] = NewStatusUpdater(c, check.Name, c.recorder)
+			c.updaters[check.Name].Start(ctx)
```

Comment:
are we guaranteed that this context is long-living, i.e. over different Sync calls? cc @mfojtik